packages/components/nodes/Etherscan/Etherscan.ts

```diff
@@ -0,0 +1,416 @@
+                        label: 'Etherscan API Key',
+                        name: 'etherscanApi',
+                    },
+                ],
+                default: 'etherscanApi',
+            },
+        ] as INodeParams[];
+        this.inputParameters = [
+            {
+                label: 'Address',
+                name: 'address',
+                type: 'string',
+                description: 'The address parameter(s) required',
+                show: {
+                    'actions.api': [
+                        'getEtherBalance',
+                        'getEtherBalanceMulti',
+                        'getContractCreatorTxHash',
+                        'getContractABI',
+                        'getContractSourceCode',
+                    ]
+                }
+            },
+            {
+                label: 'TxHash',
+                name: 'txhash',
+                type: 'string',
+                description: 'The txhash required',
+                show: {
+                    'actions.api': [
+                        'getContractExecStatus',
+                        'getTxReceiptStatus'
+                    ]
+                }
+               
+            },
+        ] as INodeParams[];
+    }
+    async run(nodeData: INodeData): Promise<INodeExecutionData[] | null> {
+        const actionData = nodeData.actions;
+        const networksData = nodeData.networks;
+        const inputParametersData = nodeData.inputParameters;
+        const credentials = nodeData.credentials;
+        if (actionData === undefined || inputParametersData === undefined || credentials === undefined || networksData === undefined) {
+            throw new Error('Required data missing');
+        }
+        // GET api
+        const api = actionData.api as string;
+        
+        // GET network
+        const network = networksData.network as NETWORK;
+        // GET credentials
+        const apiKey = credentials.apiKey as string;
+    
+        // GET address
+        const address = inputParametersData.address as string;
+        const txhash = inputParametersData.txhash as string;
+        const returnData: ICommonObject[] = [];
+        let responseData: any;
+        if (api === 'getEtherBalance') {
+            try {
+                const queryParameters = {
+                    module: 'account',
+                    action: 'balance',
+                    address,
+                    tag: 'latest',
+                    apikey: apiKey,
+                }
+                let url = '';
+                // Change url depending on network. See https://docs.etherscan.io/getting-started/endpoint-urls
+                if (network === NETWORK.MAINNET) {
+                    url = 'https://api.etherscan.io/api';
+                }
+                else if(network === NETWORK.GÖRLI){
+                    url = 'https://api-goerli.etherscan.io/api';
+                }
+                const axiosConfig: AxiosRequestConfig = {
```

Comment:
we can get rid of these duplicated codes 196:212. A good example to refer would be: https://github.com/Outerbridgeio/Outerbridge/blob/master/packages/components/nodes/PolygonScan/PolygonScan.ts